python/ray/autoscaler/_private/autoscaler.py

```diff
@@ -190,6 +195,9 @@ def _update(self):
 
             node_ip = self.provider.internal_ip(node_id)
             if node_ip in last_used and last_used[node_ip] < horizon:
+                # TODO (Alex): The node tracker will think idle nodes have
+                # failed. This is because this clause also terminated nodes
+                # that have missed heartbeats/crashed.
```

Comment:
how bad is that? so if I start my cluster with min_workers = 10 and  wait 5 minutes,  my output would show 10 failed nodes?

Reply:
Yeah pretty much. Another option could be to change "Failed nodes" to "Terminated nodes" or something like that.

Reply:
I am trying to think of how we can get into the clause. if the node is terminated or there are missed heartbeats, won't the node_ip be missing in last_used?

Reply:
I don't think we update last_used when we terminate a node. Can we merge this for now and figure out the terminated nodes issue later? This isn't a regression, and scaling down is pretty broken anyways.

Reply:
@wuisawesome can we fix this? Showing idle as failed is a pretty bad problem.

Reply:
Ok, what about we don't count crashed nodes (nodes that started correctly, but crashed after joining the cluster) as failed then? i.e. failed nodes is restricted to nodes which failed to start up properly

Reply:
Remove this TODO now?